b2sdk/session.py

```diff
@@ -8,65 +8,389 @@
+            'list_unfinished_large_files',
+            bucket_id,
+            start_file_id=start_file_id,
+            max_file_count=max_file_count,
+            prefix=prefix,
+        )
+
+    def start_large_file(self, bucket_id, file_name, content_type, file_info):
+        return self._wrap_default_token(
+            'start_large_file', bucket_id, file_name, content_type, file_info
+        )
+
+    def update_bucket(
+        self,
+        account_id,
+        bucket_id,
+        bucket_type=None,
+        bucket_info=None,
+        cors_rules=None,
+        lifecycle_rules=None,
+        if_revision_is=None,
+    ):
+        return self._wrap_default_token(
+            'update_bucket',
+            account_id,
+            bucket_id,
+            bucket_type=bucket_type,
+            bucket_info=bucket_info,
+            cors_rules=cors_rules,
+            lifecycle_rules=lifecycle_rules,
+            if_revision_is=if_revision_is,
+        )
+
+    def upload_file(
+        self, bucket_id, file_name, content_length, content_type, content_sha1, file_infos,
+        data_stream
+    ):
+        return self._wrap_token(
+            'upload_file',
+            TokenType.UPLOAD_SMALL,
+            bucket_id,
+            file_name,
+            content_length,
+            content_type,
+            content_sha1,
+            file_infos,
+            data_stream,
+        )
+
+    def upload_part(self, file_id, part_number, content_length, sha1_sum, input_stream):
+        return self._wrap_token(
+            'upload_part',
+            TokenType.UPLOAD_PART,
+            file_id,
+            part_number,
+            content_length,
+            sha1_sum,
+            input_stream,
+        )
+
+    def get_download_url_by_id(self, file_id):
+        return self.raw_api.get_download_url_by_id(self.account_info.get_download_url(), file_id)
+
+    def get_download_url_by_name(self, bucket_name, file_name):
+        return self.raw_api.get_download_url_by_name(
+            self.account_info.get_download_url(), bucket_name, file_name
+        )
+
+    def copy_file(
+        self,
+        source_file_id,
+        new_file_name,
+        bytes_range=None,
+        metadata_directive=None,
+        content_type=None,
+        file_info=None,
+        destination_bucket_id=None,
+    ):
+        return self._wrap_default_token(
+            'copy_file',
+            source_file_id,
+            new_file_name,
+            bytes_range=bytes_range,
+            metadata_directive=metadata_directive,
+            content_type=content_type,
+            file_info=file_info,
+            destination_bucket_id=destination_bucket_id,
+        )
+
+    def _wrap_default_token(self, raw_api_name, *args, **kwargs):
+        return self._wrap_token(raw_api_name, TokenType.API, *args, **kwargs)
+
+    def _wrap_token(self, raw_api_name, token_type, *args, **kwargs):
+        raw_api_method = getattr(self.raw_api, raw_api_name)
```

Comment:
this is still magic. We should be passing a method from raw_api in here instead of its name that we then map to the method dynamically...